test/reservation_test.rb

```diff
@@ -0,0 +1,49 @@
+require_relative "test_helper"
+
+describe "Reservation" do
+  before do
+    @start_date = Date.new(2020, 2, 2)
+    @end_date = @start_date + 3
+    @room = "room 3"
+    @guest_name = "Mair Heshmati"
+    @reserve = Hotel::Reservation.new(@start_date, @end_date, room: @room, guest_name: @guest_name)
+  end
+
+  describe "Initialize" do
+    it "should create an instance of Reservation" do
+      expect(@reserve).must_be_instance_of Hotel::Reservation
+    end
+
+    it "Takes in and keeps track of the start and end date, the desired room number and guest name" do
+      expect(@reserve.instance_variable_get(:@start_date)).must_equal @start_date
+      expect(@reserve.instance_variable_get(:@end_date)).must_equal @end_date
```

Comment:
In the future don't use things like `instance_variable_get` instead just put an `attr_reader` on the class.